salt/orch/update.sls

```diff
@@ -0,0 +1,121 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+
+update_modules:
+  salt.function:
+    - name: saltutil.sync_modules
+    - tgt: '*'
+    - kwarg:
+        refresh: True
+
+# Get list of masters needing reboot
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master and G@upgrade:needs', fun='network.ip_addrs', tgt_type='compound') %}
+
+{%- for master_id in masters.keys() %}
+# Call the node clean shutdown script
+{{ master_id }}-clean-shutdown:
+  salt.function:
+    - tgt: {{ master_id }}
+    - name: test.ping
+
+{{ master_id }}-clean-shutdown-kubernetes:
+  salt.state:
+    - tgt: {{ master_id }}
+    - sls:
+      - kubernetes-master.stop
+    - concurrent: False
```

Comment:
Also, I suspect the 3x clean shutdown's can be combined by just listing the 3x SLS's in the right order. I could be wrong though...